core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -596,6 +623,14 @@ private long getFlagsForStateInsertion() {
       return aggregateCompletionStage.freeze();
    }
 
+   @Override
+   public void addStore(PersistenceConfiguration persistenceConfiguration) {
+      createAndStartStores(persistenceConfiguration);
```

Comment:
Another problem is that a user may think they can change if passivation is enabled or not since they can pass a `PersistenceConfiguration`. I think this would be better if instead the argument is just a `StoreConfiguration`

Reply:
Yes, as it stands it is too generic. Probably it should allow one 1 ```StoreConfiguration``` and at the API level (REST) it should validate it is a remote-store

Reply:
> I am not a fan of reusing this method here. It has other purposes such as starting up the availability thread, which I think will be a problem here since by default that is enabled.

It was a actually the simplest refactor without changing the locking and blocking logic (```start``` was blocking before). But I get your point, it doing too much